### PR TITLE
api versioning documentation

### PR DESCRIPTION
**Pull request by isurulucky**

### CODE REVIEW

File: docs/cell-api-versioning.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Cell Api Versioning
+
+A cell defines a standard contract for the external parties to communicate with it. Hence, a it is accessible only via
```
Comment (sinthuja) [inline docs/cell-api-versioning.md]: Hence, *a* it is accessible. *a* is not required.

File: docs/cell-api-versioning.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+A cell defines a standard contract for the external parties to communicate with it. Hence, a it is accessible only via 
+the cell gateway. Each http API exposed via the cell gateway can optionally specify a version. 
+
+[Cli List Ingress](cli-reference.md#cellery-list-ingresses) command can be used to find the API versions of HTTP ingresses, 
+either for a cell image or a running cell instance. 
+
+# Exposing Apis for External Access
+
+Cell APIs can be selectively published to the Cellery global gateway to be accessed by external clients. This is done by 
+marking the relevant HTTP ingress as global, or by using a global API publisher configuration.
```
Comment (sinthuja) [inline docs/cell-api-versioning.md]: Shall we explain via sample syntax?

File: docs/cell-api-versioning.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+[Cli List Ingress](cli-reference.md#cellery-list-ingresses) command can be used to find the API versions of HTTP ingresses, 
+either for a cell image or a running cell instance. 
+
+# Exposing Apis for External Access
+
+Cell APIs can be selectively published to the Cellery global gateway to be accessed by external clients. This is done by 
+marking the relevant HTTP ingress as global, or by using a global API publisher configuration. 
+
+If only a selective set of cell APIs are to be exposed globally, each relevant ingress should be marked as global. If all 
+cell APIs are to be exposed, the global publisher configuration can be used to specify it. The global publisher configuration
```
Comment (sinthuja) [inline docs/cell-api-versioning.md]: Shall we provide a sample syntax for these cases?

File: docs/cellery-syntax.md [hunk 4]
```diff
@@ -117,11 +119,33 @@ public function build(cellery:ImageName iName) returns error? {
...
+By setting `expose` to `global`, the API will be published to the global gateway which will expose the API to external 
+traffic. In this case, the cell name and the context provided in the ingress is combined to create the full context for the 
+global API.
+
+In addition to exposing HTTP ingresses selectively, users can expose all the ingresses of a particular cell to outside by 
+using the `globalPublisher` configuration. 
+
+```ballerina
+cellery:CellImage hrCell = {
+    globalPublisher: {
```
Comment (sinthuja) [inline docs/cellery-syntax.md]: I remember, we had discussion to provide a struct for expose attribute in the component's ingress to specify how the global API will be published. But this looks bit different. Also, how can we publish all ingresses in all components under same context ?

File: docs/cellery-syntax.md [hunk 5]
```diff
@@ -84,6 +84,7 @@ A sample `HttpApiIngress` instance  with inline API definition would be as follo
 cellery:HttpApiIngress helloAPI = {
     port: 9090,
     context: "hello",
+    apiVersion: "1.0.0",
```
Comment (sinthuja) [inline docs/cellery-syntax.md]: Please explain how the APIs will be published to the global APIM in this case.

File: docs/cellery-syntax.md [hunk 6]
```diff
@@ -117,11 +119,33 @@ public function build(cellery:ImageName iName) returns error? {
...
+        apiVersion: "1.0.1",
+        context: "myorg"
+    },
+    components: {
+        hrComp: hrComponent
+    }
+};
+```
+
+Here, all the http ingresses defined in the cell will be published to the global API gateway. The context of the each
```
Comment (sinthuja) [inline docs/cellery-syntax.md]: Only the ingresses marked as global should be published to global APIM in this case.

File: docs/cellery-syntax.md (hunk 5 above)
Comment (isurulucky) [reply docs/cellery-syntax.md]: Added to the section on exposing ingresses.

**Pull request merged**